src/Development/IDE/Plugin/Completions.hs

```diff
@@ -153,8 +153,6 @@ getCompletionsLSP lsp ide
             let !position' = fromCurrentPosition mapping position
             pfix <- maybe (return Nothing) (flip VFS.getCompletionPrefix cnts) position'
```

Comment:
I looked at the code in haskell-lsp and can confirm that it is correct.
After a further look I can see that the problem has to do with position mapping.
After the a dot is added, the module doesn't have valid syntax any longer and `GetParsedModule` returns a stale result together with a mapping that advances the cursor position one character to account for the dot.

This mapped position is used here, but not with the stale parsed module content, but with current content, causing the incorrect behaviour described in my comment. Using the position from `CompletionParams` solves it and removes the need for my other changes.

Reply:
Oh yes, this is exactly what changed since the original completions code was written, as now they are always driven from stale data and then position mapping gets in the picture. But I didn't realise it needed to be used to adjust `VFS.getCompletionPrefix`, that seems a bit surprising. /cc @wz1000

Reply:
Oh yeah, nevermind, `getCContext` takes the stale `ParsedModule` so it needs the adjusted position.